assets/js/theme/product/image-gallery.js

```diff
@@ -62,19 +60,15 @@ export default class ImageGallery {
     }
 
     swapMainImage() {
+        this.easyzoom.data('easyZoom').swap(this.currentImage.mainImageUrl, this.currentImage.zoomImageUrl);
+
         this.$mainImage.attr({
             'data-zoom-image': this.currentImage.zoomImageUrl,
-        }).find('img').attr({
-            src: this.currentImage.mainImageUrl,
         });
     }
 
     setImageZoom() {
-        this.$mainImage.zoom({ url: this.$mainImage.attr('data-zoom-image'), touch: false });
-    }
-
-    destroyImageZoom() {
-        this.$mainImage.trigger('zoom.destroy');
+        this.easyzoom = $('.easyzoom').easyZoom({errorNotice: "", loadingNotice: ""});
```

Comment:
i'm thinking we can scope the call to `${'.easyzoom'`} here like `this.$mainImage.${'.easyzoom'} so it doesn't need to traverse the whole dom

Reply:
Yeah, good call.

Reply:
Indeed, there's actually no need for the .easyzoom class at all, so I just removed it entirely.

Reply:
👍